openwisp_controller/config/apps.py

```diff
@@ -61,6 +62,11 @@ def connect_signals(self):
             sender=self.vpnclient_model,
             dispatch_uid='vpnclient.post_delete',
         )
+        post_save.connect(
```

Comment:
I think that instead of using the signal, in this case we should simply override the model save method, eg:

```python
def save(self, *args, **kwargs):
    result = super().save(*args, **kwargs)
    # create
    with transaction.atomic():
        transaction.on_commit(lambda: create_vpn_dh.delay(instance.pk))
    return result
```

Signals are useful when connecting code of different modules or submodules, to achieve loose coupling but they come with the disadvantage of making the code harder to read. In this case, the advantage does not justify the price to be paid, since the `Vpn` model is supposed to know itself, we can just launch the task from the model.

[The operation must be carried out when the transaction is committed to the DB](https://docs.djangoproject.com/en/3.0/topics/db/transactions/#performing-actions-after-commit) or it may be executed before the VPN object is even created and it won't work.

This also means that some tests will have to be moved in a new `TransactionTestCase`.